libhfst/src/HfstTransducer.cc

```diff
@@ -262,11 +262,10 @@ void HfstTransducer::insert_to_alphabet(const std::string &symbol)
 
 void HfstTransducer::insert_to_alphabet(const std::set<std::string> &symbols)
 {
-    for (std::set<std::string>::const_iterator it = symbols.begin();
-         it != symbols.end(); it++)
+    for (const auto & symbol : symbols)
       {
-        HfstTokenizer::check_utf8_correctness(*it);
-        if (*it == "")
+        HfstTokenizer::check_utf8_correctness(symbol);
+        if (symbol == "")
```

Comment:
That should be `if (symbol.empty())` - not your fault, but if you're going through things anyway...

Reply:
Ok, fixed.